pkg/controller/agent/vlanconfig/controller.go

```diff
@@ -166,12 +166,15 @@ func (h Handler) setupVLAN(vc *networkv1.VlanConfig) error {
 		goto updateStatus
 	}
 	// set up VLAN
-	v = vlan.NewVlan(vc.Spec.ClusterNetwork, localAreas)
-	setupErr = v.Setup(uplink)
+	v = vlan.NewVlan(vc.Spec.ClusterNetwork)
+	if setupErr = v.Setup(uplink); setupErr != nil {
+		goto updateStatus
+	}
+	setupErr = h.AddLocalAreas(v, localAreas)
```

Comment:
whey do we ignore the setupErr here?

Reply:
We don't ignore it. The follow codes is to handle the error when updating the status.